resources/idlharness.js

```diff
@@ -3028,27 +3028,36 @@ IdlTypedef.prototype = Object.create(IdlObject.prototype);
  *      as adding objects. Do not call idl_array.test() in the setup; it is
  *      called by this function (idl_test).
  */
-function idl_test(srcs, deps, setup_func, test_name) {
+function idl_test(srcs, deps, idl_setup_func, test_name) {
     return promise_test(function (t) {
         var idl_array = new IdlArray();
         srcs = (srcs instanceof Array) ? srcs : [srcs] || [];
         deps = (deps instanceof Array) ? deps : [deps] || [];
         return Promise.all(
-            srcs.concat(deps).map(function(i) {
-                return fetch('/interfaces/' + i + '.idl').then(function(r) {
+            srcs.concat(deps).map(function(spec) {
+                return fetch('/interfaces/' + spec + '.idl').then(function(r) {
                     return r.text();
                 });
-            })).then(function(idls) {
+            }))
+            .then(function(idls) {
                 for (var i = 0; i < srcs.length; i++) {
                     idl_array.add_idls(idls[i]);
                 }
                 for (var i = srcs.length; i < srcs.length + deps.length; i++) {
                     idl_array.add_dependency_idls(idls[i]);
                 }
-                if (setup_func) {
-                    setup_func(idl_array)
-                };
-                idl_array.test();
+            })
+            .then(function() {
+                try {
```

Comment:
This try-catch wrapping is unnecessary I think, whether `idl_setup_func` throws or returns a promise that rejects, it will end up in the `catch` below.

Reply:
Done.